### PR TITLE
Fix webapi for a type of None and add missing keys

### PR DESCRIPTION
**Pull request by labrys**
...and catch None as a return value

### CODE REVIEW

**Comment by labrys** [comment]
@fernandog @duramato if you could confirm this is fixed and slap an approval on it.

**Comment by fernandog** [comment]
@labrys 

/api/****/?cmd=history

still not working:
```
{"data": [null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null], "message": "", "result": "success"}
```

File: sickbeard/webapi.py [hunk 1]
```diff
@@ -1054,7 +1054,7 @@ def convert_date(history_date):
                 ).strftime(dateTimeFormat)
 
             composite = Quality.splitCompositeStatus(cur_item.action)
-            if cur_type == statusStrings[composite.status].lower():
+            if cur_type is None or cur_type == statusStrings[composite.status].lower():
```
Comment (OmgImAlexis) [inline sickbeard/webapi.py]: Shouldn't this be not `None`? Why would you be returning stuff if it's `None`?

**Comment by fernandog** [comment]
Approved

**Comment by fernandog** [comment]
@labrys history api is missing a lot of fields!

Master:
{"data": [{"date": "2016-03-30 09:36:31", "episode": 14, "indexerid": 263365, "provider": "MoreThanTV", "quality": "1080p WEB-DL", "resource": "Marvels.Agents.of.S.H.I.E.L.D.S03E14.Watchdogs.1080p.WEB-DL.DD5.1.H.264-MAoS", "resource_path": "", "season": 3, "show_name": "Marvel's Agents of S.H.I.E.L.D.", "status": "Snatched", "tvdbid": 263365}

Develop:
{"data": [{"date": "2016-03-30 09:36:31", "indexerid": 263365, "quality": "1080p WEB-DL", "resource": "Marvels.Agents.of.S.H.I.E.L.D.S03E14.Watchdogs.1080p.WEB-DL.DD5.1.H.264-MAoS", "resource_path": "", "status": "Snatched", "tvdbid": 263365}

@duramato

File: sickbeard/webapi.py (hunk 1 above)
Comment (labrys) [inline sickbeard/webapi.py]: Because, per the API if a type isn't selected then it should return all results.

**Comment by fernandog** [comment]
now if fully fixed

**Pull request merged**